go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -134,6 +149,34 @@ func (d DoltCommitFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 	return h, err
 }
 
+func hasWorkingSetChanges(rsr env.RepoStateReader) bool {
+	if rsr.WorkingHash() != rsr.StagedHash() {
+		return true
+	}
+
+	return false
+}
+
+func hasStagedSetChanges(ctx *sql.Context, rsr env.RepoStateReader) (bool, error) {
+	root, err := rsr.HeadRoot(ctx)
```

Comment:
Make a TODO here. When you follow up with the refactor of RepoStateReader, then you should address the fact that you're returning a root object here, and shouldn't be. Just need a CommitSpec that you can get the hash from.